### ISSUE TITLE
Hyper-threading support

### ISSUE THREAD

**Issue by detritus** [vdemichev/DiaNN#512]
When trying to set the number of threads in the GUI it seems to limit itself to 50% of what I have available, I suspect it is ignoring hyper-threading.
If I manually call DIA-NN from the CLI with --threads it also seems to ignore any value higher > 50% of my processors.

This seems to occur on both Windows and Linux.

Is there any reason it is ignoring hyper-threading, is this intentional?

**Comment by detritus** [comment]
I need to look a bit further, but it might not be hyper-threading that is the issue, but instead DIA-NN might only see one physical CPU in multi-processor systems?

**Comment by detritus** [comment]
Okay, for Windows we appear to have figured out what the issue was. Hopefully this might help someone else.
By default Windows processes (inc. DIA-NN) cannot span multiple processor groups.

It is not clear what the rule is for when you may end up with multiple processor groups, in our case we have two CPUs in our Windows machine which causes Windows to create two "processor groups". We had to make a change in BIOS which fixes this, you can now clearly see DIA-NN using two processors:

![image](https://user-images.githubusercontent.com/1191435/192762743-17ae5251-65a3-4ea7-b598-624c4c898703.png)

Before:

![image](https://user-images.githubusercontent.com/1191435/192762997-c5e38cab-4e52-45c4-95a7-7e474770456f.png)

After:
![image](https://user-images.githubusercontent.com/1191435/192763029-9689a3f7-74d9-49f7-ae1c-fb64639df89d.png)

The fix for HPE/HP servers is this: https://superuser.com/questions/1633978/why-does-the-count-of-numa-nodes-influence-processor-grouping
In short, go into your system BIOS and look for "NUMA Group Size Optimization", change this to "flat"

I suspect Linux isn't affected and in our case it is just IO that is throttling the performance but I'll need to take another look.

Hope this helps someone else. For us this boosts performance by 100%, cutting my current job by approximately 80 hours. :)

**Comment by vdemichev** [comment]
@detritus , many thanks for posting a fix!
The strange thing is, DIA-NN has been initially developed on a 2-CPU PCs (some old 6-core Xeons), and it worked just fine...

**Comment by detritus** [comment]
It seems that in the Windows boot up phase it allocates processors to a "processor group", depending on 'some criteria' it depends on whether the processors go into the same group or not. An application may only run in one processor group. In our case it required a BIOS option to force Windows to put the processors into the same group.

This article contains more detail: https://learn.microsoft.com/en-us/windows/win32/procthread/processor-groups
